numpy/core/src/umath/ufunc_object.c

```diff
@@ -4387,25 +4403,36 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         return NULL;
     }
 
-    /* If an output parameter was provided, don't wrap it */
-    if (out != NULL) {
-        return (PyObject *)ret;
-    }
+    /* Wrap and return the output */
+    {
```

Comment:
I still don't like free floating blocks :)

Reply:
Would you prefer `if(true) {`? :)

I could extract this to another `_wrap_reduction_output` helper function, but I think that starts to obscure the goal here.

Usually my goal with these floating blocks is to limit the scope of the variables, to make it clear where they should be used. C99 will remove most of the reason for my use of them